x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ManageApiKeyConditionalClusterPrivilege.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.core.security.authz.privilege;
+
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.transport.TransportRequest;
+import org.elasticsearch.xpack.core.security.action.CreateApiKeyRequest;
+import org.elasticsearch.xpack.core.security.action.GetApiKeyRequest;
+import org.elasticsearch.xpack.core.security.action.InvalidateApiKeyRequest;
+import org.elasticsearch.xpack.core.security.authc.Authentication;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.BiPredicate;
+
+/**
+ * Conditional cluster privilege for managing API keys
+ */
+public final class ManageApiKeyConditionalClusterPrivilege implements ConditionalClusterPrivilege {
+
+    private static final String MANAGE_API_KEY_PATTERN = "cluster:admin/xpack/security/api_key/*";
+    private static final String CREATE_API_KEY_PATTERN = "cluster:admin/xpack/security/api_key/create";
+    private static final String GET_API_KEY_PATTERN = "cluster:admin/xpack/security/api_key/get";
+    private static final String INVALIDATE_API_KEY_PATTERN = "cluster:admin/xpack/security/api_key/invalidate";
+    private static final List<String> API_KEY_ACTION_PATTERNS = List.of(MANAGE_API_KEY_PATTERN, CREATE_API_KEY_PATTERN, GET_API_KEY_PATTERN,
+            INVALIDATE_API_KEY_PATTERN);
+
+    private final boolean restrictActionsToAuthenticatedUser;
+    private final ClusterPrivilege privilege;
+    private final BiPredicate<TransportRequest, Authentication> requestPredicate;
+
+    /**
+     * Constructor for {@link ManageApiKeyConditionalClusterPrivilege}
+     *
+     * @param actions set of API key cluster actions
+     * @param restrictActionsToAuthenticatedUser if {@code true} privileges will be restricted to current authenticated user.
+     */
+    public ManageApiKeyConditionalClusterPrivilege(Set<String> actions, boolean restrictActionsToAuthenticatedUser) {
```

Comment:
Why is any of this configurable? 
The only place it's used is in `DefaultConditionalClusterPrivilege` which calls it with fixed parameters.